src/Api/MemberApi.php

```diff
@@ -77,7 +77,79 @@ public function subscribe(string $listId, Member $member): Member
                 '/%s/lists/%s/members/%s',
                 MailChimpConfig::VERSION,
                 $listId,
-                md5($member->getEmailAddress())
+                md5(strtolower($member->getEmailAddress()))
+            ),
+            [
+                'body' => $this->serializer->serialize($member, 'json')
+            ]
+        );
+
+        /** @var $successResponse Member */
+        $successResponse = $this->serializer->deserialize((string) $response->getBody(), Member::class, 'json');
+
+        return $successResponse;
+    }
+
+    /**
+     * Re Subscribe a member
+     *
+     * @param string $listId
+     * @param Member $member
+     * @return Member
+     * @throws ValidationFailedException
+     * @throws ClientException
+     */
+    public function reSubscribe(string $listId, Member $member): Member
+    {
+        $errors = $this->validator->validate($member);
+
+        if (count($errors) > 0) {
+            throw new ValidationFailedException(get_class($member), $errors);
+        }
+
+        $member->setStatus(Member::STATUS_SUBSCRIBED);
```

Comment:
for resubscribe, we can't set user status to `STATUS_SUBSCRIBED`. This will error and mailchimp will return "Member In Compliance State"

Reply:
We only call `reSubscribe` method, if some un-subscribe user wants to subscribe to the newsletter.

 We are using `getStatus` method for check user status.

Reply:
The reason i write `getStatus` & `reSubscribe()` methods, By using  `subscribe()` method it always return `status=subscribed` where user is new or old. so we cant trace it's new user or old user.